Source/SwiftLintBuiltInRules/Rules/RuleConfigurations/FileNameConfiguration.swift

```diff
@@ -17,3 +17,20 @@ struct FileNameConfiguration: SeverityBasedRuleConfiguration {
     @ConfigurationElement(key: "require_fully_qualified_names")
     private(set) var requireFullyQualifiedNames = false
 }
+
+// MARK: - For `excluded` option
+extension FileNameConfiguration {
+    func shouldExclude(filePath: String) -> Bool {
+        let fileName = filePath.bridge().lastPathComponent
+
+        // For backwards compatibility,
+        // `excluded` can have a fileName which is invalid as regex.(e.g. "NSString+Extension.swift")
+        if excluded.contains(fileName) {
+            return true
+        }
+
+        return excluded.compactMap {
+            try? Regex("^\($0)$").firstMatch(in: filePath)
```

Comment:
I wonder if we should really match against the whole file path. Until now, only file names are checked. The regex approach should probably do the same. After all, the rule is about file names. Why would someone want to exclude file names only in specific folders?

If that's a use case, we can still add that later. For now, I think that names are sufficient.